pkg/prometheus/statefulset.go

```diff
@@ -104,25 +107,51 @@ func makeStatefulSet(p v1alpha1.Prometheus, old *v1beta1.StatefulSet, config *Co
 	return statefulset
 }
 
+func volumesInfoFromRuleConfigMaps(ruleConfigMaps []*v1.ConfigMap) ([]v1.Volume, []v1.VolumeMount, []string) {
+	volumes := make([]v1.Volume, len(ruleConfigMaps))
+	volumeMounts := make([]v1.VolumeMount, len(ruleConfigMaps))
+	ruleFileVolumeArgs := make([]string, len(ruleConfigMaps))
+
+	for i, c := range ruleConfigMaps {
+		volumeName := fmt.Sprintf("rules-%d", i)
+		ruleFilePath := configMapRuleFileFolder(i)
+
+		volumes[i] = v1.Volume{
+			Name: volumeName,
+			VolumeSource: v1.VolumeSource{
+				ConfigMap: &v1.ConfigMapVolumeSource{
+					LocalObjectReference: v1.LocalObjectReference{
+						Name: c.Name,
+					},
+				},
+			},
+		}
+		volumeMounts[i] = v1.VolumeMount{
+			Name:      volumeName,
+			ReadOnly:  true,
+			MountPath: ruleFilePath,
+		}
+		ruleFileVolumeArgs[i] = fmt.Sprintf("-volume-dir=%s", ruleFilePath)
+	}
+
+	return volumes, volumeMounts, ruleFileVolumeArgs
+}
+
 func makeEmptyConfig(name string) *v1.Secret {
 	return &v1.Secret{
 		ObjectMeta: apimetav1.ObjectMeta{
-			Name: configSecretName(name),
+			Name:   configSecretName(name),
+			Labels: managedByOperatorLabels,
+			Annotations: map[string]string{
+				"empty": "true",
```

Comment:
Yea, that's probably fine to keep for that. I assume most people `kubectl replace` the config rather than editing them. So  the annotation should stick around accidentally too often,